### PR TITLE
DR2-1982 Remove mockito from rotate password

### PR DESCRIPTION
**Pull request by MancunianSam**
This needs https://github.com/nationalarchives/da-aws-clients/pull/330 before the tests will pass

### CODE REVIEW

File: scala/lambdas/rotate-preservation-system-password/src/test/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/LambdaTest.scala [hunk 1]
```diff
@@ -1,206 +1,115 @@
...
-      .thenReturn(IO(Map("user" -> "secret")))
-    when(secretsManagerClient.getSecretValue[Map[String, String]](any[String], any[Stage])(using any[Decoder[Map[String, String]]]))
-      .thenReturn(IO(Map("user" -> "pendingSecret")))
-    val dependencies = Dependencies(secretId => IO.pure(userClient), secretId => secretsManagerClient)
+    val (secretResult, _, _) = runLambda(rotationEvent, secret, Credentials("", ""))
 
-    new Lambda().handler(rotationEvent, config, dependencies).unsafeRunSync()
-
-    verify(secretsManagerClient, times(0)).putSecretValue(any[String], any[Stage], any[Option[String]])(using any[Encoder[String]])
+    secretResult.versionToStage.size should equal(1)
```
Comment (techncl) [inline scala/lambdas/rotate-preservation-system-password/src/test/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/LambdaTest.scala:63]: Is it worth checking the password is still the same/the expected one?

**Review by techncl — COMMENTED** [review]
Looks good! Just some minor comments.

File: scala/lambdas/rotate-preservation-system-password/src/test/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/LambdaTest.scala [hunk 2]
```diff
@@ -1,206 +1,115 @@
...
+    val secret = Secret(Map("token" -> List(SecretStage(Map.empty, Pending))))
+    val (_, _, res) = runLambda(rotationEvent, secret, Credentials("", "", testSuccess = false))
 
-    new Lambda().handler(rotationEvent, config, dependencies).unsafeRunSync()
-
-    verify(userClient, times(1)).testNewPassword()
+    res.left.value.getMessage should equal("Password test failed")
   }
 
   "handler finishSecret" should "return an error if the pending secret has no version" in {
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply scala/lambdas/rotate-preservation-system-password/src/test/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/LambdaTest.scala:103]: Oh yeah, copy and paste error I think. I've changed it.

File: scala/lambdas/rotate-preservation-system-password/src/test/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/LambdaTest.scala (hunk 1 above)
Comment (MancunianSam) [reply scala/lambdas/rotate-preservation-system-password/src/test/scala/uk/gov/nationalarchives/rotatepreservationsystempassword/LambdaTest.scala:63]: It's slightly superfluous but one extra line doesn't hurt. I've added it in.

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**